### ISSUE TITLE
Hangs forever with definition line beginning as ": >" or "~ >"

### ISSUE THREAD

**Issue by ackman678** [markdown-it/markdown-it-deflist#5]
The following cases causes markdown-it using markdown-it-deflist v2.0.2 to hang forever (with no error thrown):  

Freeze case 1: 

```js
const md = require('markdown-it')().use(require('markdown-it-deflist'));
var data = `foobar
: > foo
: bar
`
md.render(data)
```

Freeze case 2: 

```js
var data = `foobar
~ > foo
~ bar
`
md.render(data)
```

But this returns okay:  

```js
var data = `foobar
: bar
: > foo
`
md.render(data)
```

And this is also okay:  

```js
var data = `foobar
: > foo
`
md.render(data)
```

So using either ":" or "~" pandoc definition syntax followed by a forward (markdown quote) caret ">" and at least one additional definition line afterwards causes markdown-it with the markdown-it-deflist module to **completely freeze**.

**Comment by rlidwka** [comment]
Fixed, thanks!

**Comment by puzrin** [comment]
Published 2.0.3 with fix & updated demo.

Thanks for report!
